### PR TITLE
Add Clamp<>

### PR DESCRIPTION
**Pull request by fdarosa2663**

### CODE REVIEW

File: functions/clamp.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#ifndef FUNCTIONS_CLAMP_H
+#define FUNCTIONS_CLAMP_H
+
+#include "int.h"
+#include "svf.h"
+
+// Usage: Clamp<F, MIN, MAX>
```
Comment (profezzorn) [inline functions/clamp.h:7]: Should be called ClampF

File: functions/clamp.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// Clamps value between MIN and MAX
+// F, A, B: INTEGER
+// return value: INTEGER
+
+template<class F, class MIN = Int<0>, class MAX = Int<32768>>
+class ClampXSVF {
```
Comment (profezzorn) [inline functions/clamp.h:13]: Needs both SVF and non-SVF versions.
(The non-svf version should do the clamp inside getInteger())

File: functions/clamp.h (hunk 1 above)
Comment (fdarosa2663) [reply functions/clamp.h]: renamed

File: functions/clamp.h (hunk 2 above)
Comment (fdarosa2663) [reply functions/clamp.h]: Added, but the SVF, non-SVF still trips me up a bit so open to improvement.

**Comment by profezzorn** [comment]
I added this code and the parts needed to make the SVF work. (hopefully, I didn't really test it.)

**Pull request closed without merging**